### PR TITLE
LPS-106787 FAKE GRADLE CACHE

### PR DESCRIPTION
**Pull request by lawrence-lee**

### CODE REVIEW

**Comment by lawrence-lee** [comment]
ci:test:sf

**Comment by lawrence-lee** [comment]
ci:test:relevant

**Comment by lawrence-lee** [comment]
ci:test:project-templates

File: modules/sdk/project-templates/project-templates-service-builder/src/main/java/com/liferay/project/templates/service/builder/internal/ServiceBuilderProjectTemplatesArgs.java [hunk 1]
```diff
@@ -33,10 +37,19 @@ public String getTemplateName() {
...
+	}
+
 	public void setDependencyInjector(String dependencyInjector) {
 		_dependencyInjector = dependencyInjector;
 	}
 
+	@Parameter(
+		description = "Set to true for add on options.", names = "--add-ons"
+	)
+	private String _addOnOptions = "false";
```
Comment (gamerson) [inline modules/sdk/project-templates/project-templates-service-builder/src/main/java/com/liferay/project/templates/service/builder/internal/ServiceBuilderProjectTemplatesArgs.java]:
```suggestion
	private String _addOns = "false";
```

File: modules/sdk/project-templates/project-templates-service-builder/src/main/java/com/liferay/project/templates/service/builder/internal/ServiceBuilderProjectTemplateCustomizer.java [hunk 2]
```diff
@@ -88,6 +111,23 @@ public void onBeforeGenerateProject(
...
+			serviceBuilderProjectTemplatesArgs.getAddOnOptions();
+
+		String liferayVersion = projectTemplatesArgs.getLiferayVersion();
+
+		if (addOnOptions.equals("true") &&
+			(liferayVersion.startsWith("7.0") ||
+			 liferayVersion.startsWith("7.1"))) {
+
+			throw new IllegalArgumentException(
+				"Add On Options is not supported in 7.0 or 7.1");
```
Comment (gamerson) [inline modules/sdk/project-templates/project-templates-service-builder/src/main/java/com/liferay/project/templates/service/builder/internal/ServiceBuilderProjectTemplateCustomizer.java]:
```suggestion
				"addOns is not supported in 7.0 or 7.1");
```

**Review by gamerson — CHANGES_REQUESTED** [review]

**Comment by lawrence-lee** [comment]
ci:stop

**Comment by lawrence-lee** [comment]
ci:test:sf

**Comment by lawrence-lee** [comment]
ci:test:relevant

**Comment by lawrence-lee** [comment]
ci:test:project-templates

**Comment by lawrence-lee** [comment]
ci:test:relevant

**Comment by lawrence-lee** [comment]
ci:test:project-templates

**Review by gamerson — APPROVED** [review]

**Comment by gamerson** [comment]
Pull request submitted to https://github.com/petershin/liferay-portal/pull/1266. <br><br>:octocat: *Sent from [GH](http://nodegh.io).*

**Pull request closed without merging**